src/main/java/bio/terra/app/usermetrics/BardClient.java

```diff
@@ -56,4 +76,56 @@ public void logEvent(AuthenticatedUserRequest userReq, BardEvent event) {
       logger.warn("Error logging event {}", event.getEvent(), e);
     }
   }
+
+  /**
+   * Syncing a user is only needed when a new token is not already cached or the cache entry has
+   * expired.
+   *
+   * @param userReq - the AuthenticatedUserRequest that represents the current user.
+   */
+  private void syncUser(AuthenticatedUserRequest userReq) {
+    String key = userReq.getToken();
+    bearerCache.computeIfAbsent(key, k -> syncProfile(userReq) ? "" : null);
```

Comment:
While this is now one line, I don't think `computeIfAbsent()` makes the operation safe for a non-synchronized map that's used across threads.

On the other hand, the failure mode is pretty low; I think the worst you get is N extra calls to `syncProfile()` depending on how many threads are trying to sync the same user. Since the map contents are ignored there's no data overwrite issue.

Reply:
Oops, nevermind I missed that it's now wrapped in SynchronizedMap.